# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.cpp
# Commit: 31fba4d1fe1d
# Full Hash: 31fba4d1fe1d16c5e0e4172bc78a8e26827849aa
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-06-13 21:41:37
# Regressor Bug: 1629788
# File Overlap Count: 1
# Description:
#   Bug 1629788 [Wayland] Add widget.wayland-dmabuf-video-textures.enabled preference, r=jhorak
#   
#   - widget.wayland-dmabuf-video-textures.enabled controls dmabuf texture backend for decoded video frames.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78293
# ==============================================================================

diff -r 9d1e4da287a6 -r 31fba4d1fe1d widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp	Sat Jun 13 10:35:40 2020 +0000
+++ b/widget/gtk/nsWaylandDisplay.cpp	Sat Jun 13 10:35:32 2020 +0000
@@ -498,6 +498,7 @@
       !StaticPrefs::widget_wayland_dmabuf_basic_compositor_enabled() &&
       !StaticPrefs::widget_wayland_dmabuf_textures_enabled() &&
 #endif
+      !StaticPrefs::widget_wayland_dmabuf_video_textures_enabled() &&
       !StaticPrefs::widget_wayland_dmabuf_webgl_enabled() &&
       !StaticPrefs::widget_wayland_dmabuf_vaapi_enabled()) {
     // Disabled by user, just quit.
@@ -535,6 +536,10 @@
 bool nsWaylandDisplay::IsDMABufBasicEnabled() { return false; }
 bool nsWaylandDisplay::IsDMABufTexturesEnabled() { return false; }
 #endif
+bool nsWaylandDisplay::IsDMABufVideoTexturesEnabled() {
+  return IsDMABufEnabled() &&
+         StaticPrefs::widget_wayland_dmabuf_video_textures_enabled();
+}
 bool nsWaylandDisplay::IsDMABufWebGLEnabled() {
   return IsDMABufEnabled() &&
          StaticPrefs::widget_wayland_dmabuf_webgl_enabled();